### PR TITLE
Add aldi.com.au

### PR DESCRIPTION
**Pull request by mlduff**
Resolves https://github.com/hhursev/recipe-scrapers/issues/1137

### CODE REVIEW

File: recipe_scrapers/aldi.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    def title(self):
+        return self.soup.find("h1").text
+
+    def category(self):
+        title = self.soup.select("a.tab-nav--link.dropdown--list--link.m-active")[
+            0
+        ].text
+        recipe_position = title.find(" Recipe")
+        return title[:recipe_position]
```
Comment (jknndy) [inline recipe_scrapers/aldi.py:28]:
```suggestion
        element = self.soup.select_one("a.tab-nav--link.dropdown--list--link.m-active")
        if element:
            title = element.text
            recipe_position = title.find(" Recipe")
            if recipe_position != -1:
                return title[:recipe_position]
```
replaced `soup.select` with `soup.select_one` to replace the list selection

File: recipe_scrapers/aldi.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            total_time += self.prep_time()
+        except ElementNotFoundInHtml:
+            pass
+
+        try:
+            total_time += self.cook_time()
+        except ElementNotFoundInHtml:
+            pass
+
+        return total_time
```
Comment (jknndy) [inline recipe_scrapers/aldi.py:49]:
```suggestion
        return (self.prep_time() or 0) + (self.cook_time() or 0)
```
simplified by using the `or` operator to handle the possible none values. let me know if there was additional reasoning for the original code that required this work around!

File: recipe_scrapers/aldi.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    def image(self):
+        figure = self.soup.find(
+            "figure", {"class": "csc-textpic-image csc-textpic-last"}
+        )
+        if not figure:
+            return None
+        image = figure.find("img")
+        if not image:
+            return None
+        return image.get("src")
```
Comment (jknndy) [inline recipe_scrapers/aldi.py:64]:
```suggestion
        figure = self.soup.find("figure", {"class": "csc-textpic-image csc-textpic-last"})
        if figure:
            image = figure.find("img")
            if image:
                return image.get("src")
        return None
```
simplified the nested checks into a single `if` statement.

File: recipe_scrapers/aldi.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        for li in list_element.find_all("li"):
+            ingredients.append(li.text.strip())
+        return ingredients
+
+    def instructions(self):
+        list_element = self.soup.find("ol")
+        instructions = []
+        for li in list_element.find_all("li"):
+            instructions.append(li.text.strip())
+        return "\n".join(instructions)
```
Comment (jknndy) [inline recipe_scrapers/aldi.py:79]:
```suggestion
        return "\n".join(li.text.strip() for li in list_element.find_all("li"))
```
changed to use list comprehension to create the instructions list

File: recipe_scrapers/aldi.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        parts = []
+        for br in br_tags:
+            next_sibling = br.next_sibling
+            while next_sibling:
+                text = next_sibling.text
+                if text:
+                    parts.append(text)
+                next_sibling = next_sibling.next_sibling
+
+        return " ".join(parts)
```
Comment (jknndy) [inline recipe_scrapers/aldi.py:97]:
```suggestion
        label_element = self.soup.find("b", string=label)
        if label_element:
            parts = [sibling.strip() for sibling in label_element.find_next_siblings(string=True) if sibling.strip()]
            return " ".join(parts)
        return None
```

changed to use list comprehension to gather the text from the label's next siblings & added `.strip` to cleanup the outputs.

File: recipe_scrapers/aldi.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
+# mypy: allow-untyped-defs
+import re
+
+from ._abstract import AbstractScraper
+from ._exceptions import ElementNotFoundInHtml
```
Comment (jknndy) [inline recipe_scrapers/aldi.py:5]:
```suggestion
```
related to `total_time` change

**Review by jknndy — CHANGES_REQUESTED** [review]

**Comment by jknndy** [comment]
Also don't forget the readme entry!

**Comment by mlduff** [comment]
> Also don't forget the readme entry!

If only there was a test for that 😉

**Review by jknndy — APPROVED** [review]

**Comment by jayaddison** [comment]
Thank you @mlduff!

**Pull request merged**